File: backend/src/main/java/com/example/upload/domain/post/post/dto/PageDto.java
package com.example.upload.domain.post.post.dto;

import com.example.upload.domain.post.post.entity.Post;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.data.domain.Page;
import org.springframework.lang.NonNull;

import java.util.List;

@Getter
@AllArgsConstructor
public class PageDto {

    @NonNull
    List<PostDto> items;
    @NonNull
    int totalPages;
    @NonNull
    int totalItems;
    @NonNull
    int currentPageNo;
    @NonNull
    int pageSize;

    public PageDto(Page<Post> postPage) {

        this.items = postPage.getContent().stream()
                .map(PostDto::new)
                .toList();

        this.totalPages = postPage.getTotalPages();
        this.totalItems = (int) postPage.getTotalElements();
        this.currentPageNo = postPage.getNumber() + 1;
        this.pageSize = postPage.getSize();
    }

}

File: backend/src/main/java/com/example/upload/domain/post/post/dto/PageDto.kt
package com.example.upload.domain.post.post.dto

import com.example.upload.domain.post.post.entity.Post
import org.springframework.data.domain.Page

class PageDto(
    var items: List<PostDto>,
    var totalPages: Int,
    var totalItems: Int,
    var currentPageNo: Int,
    var pageSize: Int
) {
    constructor(postPage: Page<Post>): this(
        items = postPage.content.map { PostDto(it) },
        totalPages= postPage.totalPages,
        totalItems= postPage.totalElements.toInt(),
        currentPageNo= postPage.number + 1,
        pageSize= postPage.size
    )
}
